File: Plugins/Aspose_Cells_Java_for_Ruby/lib/asposecellsjava/freezepanes.rb
module Asposecellsjava
  module FreezePanes
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(__FILE__))) + '/data/'
        
        # Instantiating a Workbook object by excel file path
        workbook = Rjb::import('com.aspose.cells.Workbook').new(data_dir + 'Book1.xls')
        
        # Accessing the first worksheet in the Excel file
        worksheets = workbook.getWorksheets()
        worksheet = worksheets.get(0)

        # Applying freeze panes settings
        worksheet.freezePanes(3,2,3,2)
        
        # Saving the modified Excel file in default (that is Excel 2003) format
        workbook.save(data_dir + "output.xls")

        puts "Apply freeze panes settings, please check the output file."
    end
  end
end

File: Examples/src/main/java/com/aspose/cells/examples/worksheets/display/FreezePanes.java
package com.aspose.cells.examples.worksheets.display;

import com.aspose.cells.*;
import com.aspose.cells.examples.Utils;

public class FreezePanes {

	public static void main(String[] args) throws Exception {
		// ExStart:1
		// The path to the documents directory.
		String dataDir = Utils.getDataDir(FreezePanes.class);

		// Instantiating a Excel object by excel file path
		Workbook workbook = new Workbook(dataDir + "book.xls");

		// Accessing the first worksheet in the Excel file
		WorksheetCollection worksheets = workbook.getWorksheets();
		Worksheet worksheet = worksheets.get(0);

		// Applying freeze panes settings
		worksheet.freezePanes(3, 2, 3, 2);

		// Saving the modified Excel file in default format
		workbook.save(dataDir + "book.out.xls");

		// Print Message
		System.out.println("Panes freeze successfull.");
		// ExEnd:1
	}
}
